test/test_container_example_apps.py

```diff
@@ -0,0 +1,176 @@
+import pytest
+import re
+
+from pathlib import Path
+
+from container_ci_suite.container_lib import ContainerTestLib
+from container_ci_suite.engines.podman_wrapper import PodmanCLIWrapper
+from container_ci_suite.engines.container import ContainerImage
+
+from conftest import VARS
+
+test_app = VARS.TEST_DIR / "test-app"
+perl_test_app = VARS.TEST_DIR / "perl-test-app"
+start_hook_test_app = VARS.TEST_DIR / "start-hook-test-app"
+
+
+def build_s2i_app(app_path: Path) -> ContainerTestLib:
+    """
+    Build S2I Container based on the app_path input
+    :param: app_path: path to example directory
+    :return new instance of ContainerTestLib
+    """
+    container_lib = ContainerTestLib(VARS.IMAGE_NAME)
+    app_name = app_path.name
+    s2i_app = container_lib.build_as_df(
+        app_path=app_path,
+        s2i_args="--pull-policy=never",
+        src_image=VARS.IMAGE_NAME,
+        dst_image=f"{VARS.IMAGE_NAME}-{app_name}"
+    )
+    s2i_app.set_new_image(image_name=f"{VARS.IMAGE_NAME}-{app_name}")
+    return s2i_app
+
+
+class TestNginxExampleAppContainer:
+
+    def setup_method(self):
+        self.s2i_app = build_s2i_app(start_hook_test_app)
+
+    def teardown_method(self):
+        self.s2i_app.cleanup()
+
+    def test_run_app_test(self):
+        cid_file_name = self.s2i_app.app_name
+        # Create container with --user 10001
+        assert self.s2i_app.create_container(cid_file_name=cid_file_name, container_args="--user 10001")
+        # Wait till container does not start
+        assert ContainerImage.wait_for_cid(cid_file_name=cid_file_name)
+        cid = self.s2i_app.get_cid(cid_file_name=cid_file_name)
+        assert cid
+        cip = self.s2i_app.get_cip(cid_file_name=cid_file_name)
+        assert cip
+        command = PodmanCLIWrapper.podman_get_file_content(
+            cid_file_name=cid, filename="/opt/app-root/etc/nginx.d/default.conf"
+        )
+        # Checks if nginx configuration contains resolver
+        assert re.search("resolver", command)
+        # Checks if nginx configuration DO NOT container "DNS_SERVER"
+        assert not re.search("DNS_SERVER", command)
```

Comment:
`command` is not a very good descriptive name for this variable.

```suggestion
        nginx_config = PodmanCLIWrapper.podman_get_file_content(
            cid_file_name=cid, filename="/opt/app-root/etc/nginx.d/default.conf"
        )
        # Checks if nginx configuration contains resolver
        assert re.search("resolver", nginx_config)
        # Checks if nginx configuration DO NOT container "DNS_SERVER"
        assert not re.search("DNS_SERVER", nginx_config)
```